File: java_solutions/1309.decrypt-string-from-alphabet-to-integer-mapping.java
/*
 * @lc app=leetcode id=1309 lang=java
 *
 * [1309] Decrypt String from Alphabet to Integer Mapping
 */

class Solution {
    public String freqAlphabets(String s) {
        return java.util.regex.Pattern.compile("(\\d{2}#|\\d)")
            .matcher(s)
            .results()
            .map(m -> m.group())
            .map(v -> (char) ('a' + Integer.parseInt(v.replace("#", "")) - 1))
            .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append)
            .toString();
    }
}

File: kotlin_solutions_1/1309.decrypt-string-from-alphabet-to-integer-mapping.kt
/*
 * @lc app=leetcode id=1309 lang=kotlin
 *
 * [1309] Decrypt String from Alphabet to Integer Mapping
 */
class Solution { fun freqAlphabets(s: String): String = Regex("""\d{2}#|\d""").findAll(s).map { ('a' + it.value.removeSuffix("#").toInt() - 1) }.joinToString("") }